kratos/processes/calculate_embedded_nodal_variable_from_skin_process.h

```diff
@@ -0,0 +1,746 @@
+        mpFindIntersectedGeometricalObjectsProcess->FindIntersections();
+    }
+
+    void ClearIntersections()
+    {
+        mpFindIntersectedGeometricalObjectsProcess->Clear();
+    }
+
+    const Vector SetDistancesVector(ModelPart::ElementIterator ItElem)
+    {
+        auto &r_geom = ItElem->GetGeometry();
+        Vector nodal_distances(r_geom.PointsNumber());
+
+        if (mLevelSetType == "continuous"){
+            // Continuous nodal distance function case
+            for (unsigned int i_node = 0; i_node < r_geom.PointsNumber(); ++i_node) {
+                nodal_distances[i_node] = r_geom[i_node].FastGetSolutionStepValue(DISTANCE);
+            }
+        } else if (mLevelSetType == "discontinuous") {
+            // Discontinuous elemental distance function case
+            nodal_distances = ItElem->GetValue(ELEMENTAL_DISTANCES);
+        } else {
+            KRATOS_ERROR << "Level set type must be either 'continuous' or 'discontinuous'. Got " << mLevelSetType;
+        }
+
+        return nodal_distances;
+    }
+
+    inline bool IsSplit(const Vector &rDistances)
+    {
+        unsigned int n_pos = 0, n_neg = 0;
+
+        for (double dist : rDistances) {
+            if(dist >= 0) {
+                ++n_pos;
+            } else {
+                ++n_neg;
+            }
+        }
+
+        if (n_pos > 0 && n_neg > 0) {
+            return true;
+        }
+
+        return false;
+    }
+
+	int ComputeEdgeIntersection(
+		const Element::GeometryType& rIntObjGeometry,
+		const Element::NodeType& rEdgePoint1,
+		const Element::NodeType& rEdgePoint2,
+		Point& rIntersectionPoint)
+	{
+		int intersection_flag = 0;
+		const auto work_dim = rIntObjGeometry.WorkingSpaceDimension();
+		if (work_dim == 2){
+			intersection_flag = IntersectionUtilities::ComputeLineLineIntersection<Element::GeometryType>(
+				rIntObjGeometry, rEdgePoint1.Coordinates(), rEdgePoint2.Coordinates(), rIntersectionPoint.Coordinates());
+		} else if (work_dim == 3){
+			intersection_flag = IntersectionUtilities::ComputeTriangleLineIntersection<Element::GeometryType>(
+				rIntObjGeometry, rEdgePoint1.Coordinates(), rEdgePoint2.Coordinates(), rIntersectionPoint.Coordinates());
+		} else {
+			KRATOS_ERROR << "Working space dimension value equal to " << work_dim << ". Check your skin geometry implementation." << std::endl;
+		}
+
+        return intersection_flag;
+    }
+
+    void AddEdgeNodes(
+        const Geometry<Node<3>> &rEdgeGeometry,
+        ModelPart &rModelPart)
+    {
+        // Loop the edge nodes
+        for (std::size_t i = 0; i < 2; ++i) {
+            auto p_i_node = rEdgeGeometry(i);
+            // Check if the node has been already added
+            if (!p_i_node->Is(VISITED)) {
+                p_i_node->Set(VISITED, true);
+                const auto new_id = p_i_node->Id();
+                Node<3>::Pointer p_new_node = rModelPart.CreateNewNode(new_id, *p_i_node);
```

Comment:
```suggestion
                rModelPart.CreateNewNode(p_i_node->Id(), *p_i_node);
```

I would be careful with auto for ints and doubles

Reply:
Changed. Just for my understanding, why is it dangerous?

Reply:
auto i=5 is an int, but auto i=5.0 is a double (if I am not wrong) 
I had a nasty bug with this in the past, thats why I try to not use it for those types
For complex types ofc, but I so far I haven’t had a good usecase dor double/int

Reply:
Makes sense. I'll go through the code one more time to check this again. Thanks for the hint!